packages/hint-compat-api/src/helpers/compat-api.ts

```diff
@@ -6,36 +6,40 @@
 const bcd: CompatData = require('mdn-browser-compat-data');
 
 import { browserVersions } from './normalize-version';
-import { BrowserSupportCollection, MDNTreeFilteredByBrowsers } from '../types';
+import { BrowserSupportCollection, MDNTreeFilteredByBrowsers, BrowserVersions } from '../types';
 import { CompatData, CompatStatement, SupportStatement, SimpleSupportStatement, Identifier } from '../types-mdn.temp'; // Temporal
+import { userBrowsers } from './get-user-browsers';
+import { HintContext } from 'hint/dist/src/lib/hint-context';
 
 type CompatNamespace = 'css' | 'javascript' | 'html';
 
 export class CompatApi {
     public compatDataApi: MDNTreeFilteredByBrowsers;
-    private isCheckingNotBroadlySupported = false;
+    private readonly isCheckingNotBroadlySupported: boolean;
+    private readonly mdnBrowsersCollection: BrowserSupportCollection;
 
-    public constructor(namespaceName: CompatNamespace, browsers: BrowserSupportCollection, isCheckingNotBroadlySupported = false) {
+    public constructor(namespaceName: CompatNamespace, context: HintContext, isCheckingNotBroadlySupported = false) {
+        this.mdnBrowsersCollection = userBrowsers.convert(context.targetedBrowsers);
         this.isCheckingNotBroadlySupported = isCheckingNotBroadlySupported;
         this.compatDataApi = bcd[namespaceName];
-        this.compatDataApi = this.filterCompatDataByBrowsers(browsers);
+        this.compatDataApi = this.filterCompatDataByBrowsers();
     }
 
-    private filterCompatDataByBrowsers(browsers: BrowserSupportCollection): MDNTreeFilteredByBrowsers {
+    private filterCompatDataByBrowsers(): MDNTreeFilteredByBrowsers {
```

Comment:
yep, I do not have context about this project, probably is correct, but I do not like the name of this function, I think it will do a filter action, but it seems has a 'macher' behavior.